common/src/main/java/com/wynntils/features/combat/QuickCastFeature.java

```diff
@@ -124,17 +126,30 @@ private void tryCastSpell(SpellUnit a, SpellUnit b, SpellUnit c) {
                 return;
             }
 
+            // First check if the character is an archer or not in case CharacterModel failed to parse correctly
             Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(heldItem, GearItem.class);
 
             if (gearItemOpt.isEmpty()) {
-                sendCancelReason(Component.translatable("feature.wynntils.quickCast.notAWeapon"));
-                return;
-            } else if (!gearItemOpt.get().meetsActualRequirements()) {
+                Optional<CraftedGearItem> craftedGearItemOpt = Models.Item.asWynnItem(heldItem, CraftedGearItem.class);
+
+                if (craftedGearItemOpt.isEmpty()) {
+                    sendCancelReason(Component.translatable("feature.wynntils.quickCast.notAWeapon"));
+                    return;
+                }
+
+                isArcher = craftedGearItemOpt.get().getRequiredClass() == ClassType.ARCHER;
```

Comment:
Don't we have a common interface for gear item and crafted gear items that can supply getRequiredClass()? If not, maybe we should create one? "ClassableItemProperty" or whatever...

Reply:
Not currently but I can do it here if that would be better

Reply:
Not added it to UnknownGearItem as we don't have any class info there

Reply:
UnknownGearItem does have `requirements`, containing class info.

Reply:
So it does, I missed that 😅